solar_system.rb

```diff
@@ -0,0 +1,211 @@
+
+#SETTING CLASSES
+class Planet
+  attr_reader :name, :year_length, :distance_from_sun, :type, :moons,
+  :inhabitants
+
+  def initialize(name, year_length, distance_from_sun, type, moons, inhabitants)
+    @name = name
+    @year_length = year_length
+    @distance_from_sun = distance_from_sun
+    @type = type
+    @moons = moons
+    @inhabitants = inhabitants
+  end
+
+  def planet_all_info
+    output = ""
+    output << "\n    Planet name: #{name}
+    Length of year in earth-days: #{year_length}
+    Distance from the sun in some units: #{distance_from_sun}
+    Class type: #{type}
+    Number of Moons: #{moons}
+    Home to: #{inhabitants}\n "
+    return output
+  end
+end
+
+class SolarSystem
+  attr_accessor :planets
+
+  def initialize(planets)
+    @planets = planets
+  end
+
+  def add_planet(new_planet)
+    @planets << new_planet
+  end
+
+  def ss_name_list
+    i=1
+    name_list = ""
+    planets.each do |planets|
+      name_list<< "#{i}. #{planets.name}\n"
+      i +=1
+    end
+    return name_list
+  end
+
+end
+
+#DEFINING METHODS
+#is the user input a y or a n?  if not, keep cylcing through
+def y_or_n(input)
+  until input == "Y" || input == "N"
+    puts "Say what? Please enter a Y or an N"
+    input = gets.chomp.upcase
+  end
+  return input
+end
+
+#Checks that the user-inputed planet name is the name of a planet
+#in this solar system. Maybe there is a way to do this without
+#building an array of names? If it is not a name then the program
+#asks the user for a name until they can manage to enter a name.
+def input_is_name (my_solar_system, input)
+  name_array = []
```

Comment:
Since it takes a solar system as a parameter, this might be a good candidate to be an instance method of class `SolarSystem`.

For checking the name without creating an array, why not:

```ruby
def input_is_name(my_solar_system, input)
  my_solar_system.planets.each do |planet|
    if planet.name == input
      return true
    end
  end
  return false
end
```